core-bundle/src/Command/UserCreateCommand.php

```diff
@@ -0,0 +1,328 @@
+
+        $question->setValidator($callback);
+
+        /** @var QuestionHelper $helper */
+        $helper = $this->getHelper('question');
+
+        return $helper->ask($input, $output, $question);
+    }
+
+    private function askChoice(string $label, array $options, InputInterface $input, OutputInterface $output): string
+    {
+        $question = new ChoiceQuestion($label, $options);
+        $question->setAutocompleterValues($options);
+
+        /** @var QuestionHelper $helper */
+        $helper = $this->getHelper('question');
+
+        return $helper->ask($input, $output, $question);
+    }
+
+    private function askMultipleChoice(
+        string $label,
+        array $options,
+        InputInterface $input,
+        OutputInterface $output
+    ): array {
+        $question = new ChoiceQuestion($label, $options);
+        $question->setAutocompleterValues($options);
+        $question->setMultiselect(true);
+
+        /** @var QuestionHelper $helper */
+        $helper = $this->getHelper('question');
+
+        return $helper->ask($input, $output, $question);
+    }
+
+    private function getGroups(): array
+    {
+        $this->framework->initialize();
+
+        /** @var UserGroupModel $userGroupModel */
+        $userGroupModel = $this->framework->getAdapter(UserGroupModel::class);
+        $groups = $userGroupModel->findAll();
+
+        if (null === $groups) {
+            return [];
+        }
+
+        return $groups->fetchEach('name');
+    }
+
+    private function persistUser(
+        string $username,
+        string $name,
+        string $email,
+        string $password,
+        string $language,
+        bool $isAdmin = false,
+        array $groups = null,
+        bool $pwChange = false
+    ): void {
+        $time = time();
+        $hash = $this->encoderFactory->getEncoder(BackendUser::class)->encodePassword($password, null);
+
+        $this->connection->createQueryBuilder()
+            ->insert('tl_user')
+            ->values(
+                [
+                    'tstamp' => '?',
+                    'name' => '?',
+                    'email' => '?',
+                    'username' => '?',
+                    'password' => '?',
+                    'language' => '?',
+                    'backendTheme' => '?',
+                    'admin' => '?',
+                    'pwChange' => '?',
+                    'dateAdded' => '?',
+                    'groups' => '?',
+                ]
+            )
+            ->setParameters(
+                [
+                    $time,
+                    $name,
+                    $email,
+                    $username,
+                    $hash,
+                    $language,
+                    'flexible',
+                    $isAdmin,
+                    $pwChange,
+                    $time,
+                    !$isAdmin && !empty($groups) ? serialize(array_map('strval', $groups)) : '',
+                ]
+            )
+            ->execute()
```

Comment:
You can probably use `$this->connection->insert()` (see [doctrine docs](https://www.doctrine-project.org/projects/doctrine-dbal/en/2.10/reference/data-retrieval-and-manipulation.html#insert)) with an associative array for better readability.

Reply:
Thank you @m-vo! 3c6c6a6